test/integration/smoke/test_retrieve_diagnostics.py

```diff
@@ -0,0 +1,235 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
As of structure I think this looks good, let me get back on this when I pick this up for testing and possibly point any missing scenario.